ui/app/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js

```diff
@@ -99,26 +102,93 @@ export default class PermissionPageContainerContent extends PureComponent {
     )
   }
 
-  render () {
-    const { domainMetadata, redirect, permissionRejected, selectedIdentities } = this.props
+  getAccountDescriptor (identity) {
+    return `${identity.label} (...${identity.address.slice(identity.address.length - 4)})`
+  }
+
+  renderAccountTooltip (textContent) {
+    const { selectedIdentities } = this.props
+    const { t } = this.context
+
+    return (
+      <Tooltip
+        key="all-account-connect-tooltip"
+        position="bottom"
+        wrapperClassName="permission-approval-container__bold-title-elements"
+        html={(
+          <div style={{ display: 'flex', flexDirection: 'column' }}>
+            { selectedIdentities.slice(0, 6).map((identity, index) => {
+              return (
+                <div key={ `tooltip-identity-${index}` }>
+                  { this.getAccountDescriptor(identity) }
+                </div>
+              )
+            }) }
+            { selectedIdentities.length > 6
+              ? t('plusXMore', [ selectedIdentities.length - 6 ])
+              : null
+            }
+          </div>
+        )}
+      >{ textContent }
+      </Tooltip>
+    )
+  }
+
+  renderTitleSubstituteText (name, key = 'title-substitute-text') {
+    return (
+      <div
+        key={key}
+        className="permission-approval-container__bold-title-elements"
+      >
+        { name }
+      </div>
+    )
+  }
+
+  getTitleArgs () {
+    const { domainMetadata, redirect, permissionRejected, selectedIdentities, allIdentitiesSelected } = this.props
     const { t } = this.context
 
-    let titleArgs
     if (redirect && permissionRejected) {
-      titleArgs = [ 'cancelledConnectionWithMetaMask' ]
+      return [ 'cancelledConnectionWithMetaMask' ]
     } else if (redirect) {
-      titleArgs = [ 'connectingWithMetaMask' ]
+      return [ 'connectingWithMetaMask' ]
     } else if (domainMetadata.extensionId) {
-      titleArgs = [ 'externalExtension', [domainMetadata.extensionId] ]
+      return [ 'externalExtension', [domainMetadata.extensionId] ]
+    } else if (allIdentitiesSelected) {
+      return [
+        'connectToAll',
+        [
+          <div
```

Comment:
Nit: you could make this a `span` instead, which would mean no longer having to change the display to `inline-block`. Generally I think it makes sense to prefer `span` to `div` for arbitrary containers of inline content.

Reply:
The same goes for the other `div`s with this class